### PR TITLE
TSK-1306: Fixed problem with domain switching in classifications

### PR DESCRIPTION
**Pull request by Tristan2357**
https://sonarcloud.io/dashboard?branch=TSK-1306&id=Tristan2357_taskana

### CODE REVIEW

File: web/src/app/shared/services/domain/domain.service.ts [hunk 1]
```diff
@@ -64,9 +64,7 @@ export class DomainService {
   }
 
   switchDomain(value: string) {
-    this.requestInProgressService.setRequestInProgress(true);
     this.selectDomain(value);
-    this.router.navigate([this.getNavigationUrl()]);
```
Comment (mustaphazorgati) [inline web/src/app/shared/services/domain/domain.service.ts:69]: what's the reason for this change?

Comment (Tristan2357) [reply web/src/app/shared/services/domain/domain.service.ts:69]: Switching the domain should not affect the url, therefore the second call is not needed.
The selectDomain call is synchronous and should not need a spinner. The actual loading of the classifications or the workbaskets from another domain maybe should show a spinner, but in my opinion, this is not the responsibilitie of the domain service. If we were to start the spinner in this service, it should be understandable where it will stop it. this was not the case, and in most cases, the spinner was not stopped again, which caused the domainSwitching to fail in the first place.

Comment (Tristan2357) [reply web/src/app/shared/services/domain/domain.service.ts:69]: There would be solutions with the call for inProgress still there, would this be prefered? If yes this should be easily possible. 
Maybe i should start the inProgress at the inner calls? Or should i continue the current pattern?

Comment (mustaphazorgati) [reply web/src/app/shared/services/domain/domain.service.ts:69]: Let's talk about this. Maybe we need to refactor this service in order to get rid of this behaviour.

Currently this change leaves code fragments everywhere where `domainChangedComplete` is used.

Furthermore we have a behaviour change: previously changing the domain unselected the active classification / workbasket. This is not the case anymore
